chainstate/test-framework/src/framework_builder.rs

```diff
@@ -67,7 +67,7 @@ impl TestFrameworkBuilder {
             max_db_commit_attempts: Default::default(),
             max_orphan_blocks: Default::default(),
             min_max_bootstrap_import_buffer_sizes: Default::default(),
-            tx_index_enabled: true.into(),
+            tx_index_enabled: rng.gen_bool(0.5).into(),
```

Comment:
I am somewhat uncomfortable that something called `new` is not deterministic. Not quite obvious from call sites but not sure what a better approach could be